pkg/reconciler/v1alpha1/service/service_test.go

```diff
@@ -427,79 +332,192 @@ func TestNew(t *testing.T) {
-		RunLatest: &v1alpha1.RunLatestType{Configuration: configSpec},
-	}, conditions...)
+func MarkRouteReady(s *v1alpha1.Service) {
+	s.Status.PropagateRouteStatus(v1alpha1.RouteStatus{
+		Conditions: []duckv1alpha1.Condition{{
+			Type:   "Ready",
+			Status: "True",
+		}},
+	})
+}
+
+func MarkRouteFailed(reason, message string) ServiceOption {
+	return func(s *v1alpha1.Service) {
+		s.Status.PropagateRouteStatus(v1alpha1.RouteStatus{
+			Conditions: []duckv1alpha1.Condition{{
+				Type:    "Ready",
+				Status:  "False",
+				Reason:  reason,
+				Message: message,
+			}},
+		})
+	}
+}
+
+func MarkConfigReady(s *v1alpha1.Service) {
+	s.Status.PropagateConfigurationStatus(v1alpha1.ConfigurationStatus{
+		Conditions: []duckv1alpha1.Condition{{
+			Type:   "Ready",
+			Status: "True",
+		}},
+	})
+}
+
+func MarkConfigFailed(reason, message string) ServiceOption {
+	return func(s *v1alpha1.Service) {
+		s.Status.PropagateConfigurationStatus(v1alpha1.ConfigurationStatus{
+			Conditions: []duckv1alpha1.Condition{{
+				Type:    "Ready",
+				Status:  "False",
+				Reason:  reason,
+				Message: message,
+			}},
+		})
+	}
+}
+
+func MarkAllReady(s *v1alpha1.Service) {
+	MarkRouteReady(s)
+	MarkConfigReady(s)
+}
+
+func WithRunLatest(s *v1alpha1.Service) {
+	s.Spec = v1alpha1.ServiceSpec{
+		RunLatest: &v1alpha1.RunLatestType{
+			Configuration: configSpec,
+		},
+	}
 }
 
-func svcPin(name, namespace string, conditions ...duckv1alpha1.Condition) *v1alpha1.Service {
-	return svc(name, namespace, v1alpha1.ServiceSpec{
-		Pinned: &v1alpha1.PinnedType{RevisionName: "pinned-0001", Configuration: configSpec},
-	}, conditions...)
+func WithPin(s *v1alpha1.Service) {
+	s.Spec = v1alpha1.ServiceSpec{
+		Pinned: &v1alpha1.PinnedType{
+			RevisionName:  "pinned-0001",
+			Configuration: configSpec,
+		},
+	}
 }
 
-func svcRelease(name, namespace string, conditions ...duckv1alpha1.Condition) *v1alpha1.Service {
-	return svc(name, namespace, v1alpha1.ServiceSpec{
-		Release: &v1alpha1.ReleaseType{Revisions: []string{"release-00001", "release-00002"}, RolloutPercent: 49, Configuration: configSpec},
-	}, conditions...)
+func WithRelease(s *v1alpha1.Service) {
+	s.Spec = v1alpha1.ServiceSpec{
+		Release: &v1alpha1.ReleaseType{
+			Revisions: []string{
+				"release-00001",
+				"release-00002",
+			},
+			RolloutPercent: 49,
+			Configuration:  configSpec,
+		},
+	}
 }
 
-func svcManual(name, namespace string, conditions ...duckv1alpha1.Condition) *v1alpha1.Service {
-	return svc(name, namespace, v1alpha1.ServiceSpec{
+func WithManual(s *v1alpha1.Service) {
+	s.Spec = v1alpha1.ServiceSpec{
 		Manual: &v1alpha1.ManualType{},
-	}, conditions...)
+	}
 }
 
-func mustMakeConfig(t *testing.T, svc *v1alpha1.Service) *v1alpha1.Configuration {
-	cfg, err := resources.MakeConfiguration(svc)
+type ConfigOption func(*v1alpha1.Configuration)
+
+func config(name, namespace string, so ServiceOption, co ...ConfigOption) *v1alpha1.Configuration {
```

Comment:
When seeing `config("config", "namespace", WithPin)` I couldn't infer how WithPin was manipulating the config - since I thought it modified a service.

If you only allow `ConfigOption` to be used I could see an alternative 

```go
config("name", "namespace", WithPinnedSpec("service", "namespace"))
```

where `WithPinnedSpec` may be defined like

```go
func WithPinnedSpec(serviceName, serviceNamespace, so ....ServiceOption) *v1alpha1.Configuration
```

Maybe the `namespace` could be omitted